lib/openshift/catalog_source.rb

```diff
@@ -9,5 +9,30 @@ def ready?(user:, quiet: false, cached: false)
   		res[:success] = res[:parsed]["status"]["connectionState"]["lastObservedState"] == "READY"
   		return res
   	end
+
+  	def endpoint(user: nil, quiet: false, cached: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      return rr.dig('spec', 'image')
+  	end
+
+  	def registrypollinterval(user: nil, quiet: false, cached: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      return rr['spec'].has_key?('updateStrategy') ? rr.dig('spec', 'updateStrategy', 'registryPoll', 'interval') : '-'	
+  	end
+
+  	def publisher(user: nil, quiet: false, cached: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      return rr['spec'].has_key?('publisher') ? rr.dig('spec', 'publisher') : '-'	
+  	end
+
+  	def status(user: nil, quiet: false, cached: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      return rr.dig('status', 'connectionState', 'lastObservedState')  		
+  	end
+
+  	def displayname(user: nil, quiet: false, cached: false)
```

Comment:
I think we can cache most of the resources, except status, by default

Reply:
@liangxia  Good idea, let me update

Reply:
updated, PTAL @liangxia